CMM-988 android empty states use outdated visual styles

adalpari opened the pull request:
## Description
Modern Reader empty states with Material 3 icons. 
Replaced outdated PNG illustrations with clean Material 3 vector icons for better theming and scalability.

I verified other use cases to detect old illustration usages, but it looks like this is the mainly used one. I used a "tag" icon, though, for the `ShowNoFollowedTagsUiState`

| Image                                | Type                | Primary Use                   | States Count      |
  |--------------------------------------|---------------------|-------------------------------|-------------------|
  | illustration_reader_empty            | PNG (Multi-density) | Main empty state illustration | ~15 states        |
  | ic_wifi_off_24px                     | Vector              | Network/offline errors        | 5 states          |
  | ic_notice_48dp                       | PNG (Multi-density) | Error notices                 | 1 state           |
  | img_illustration_empty_results_216dp | Vector              | Empty results (fallback)      | Tool preview only |

It also fixes Fixes CMM-995, opened here: https://github.com/wordpress-mobile/WordPress-Android/pull/22367#issuecomment-3576127744

## Testing instructions
1. Log in to the app with a new self-hosted account
2. Go to the reader
3. Tap on "Following"
- [ ] Verify the empty state is not using the old illustration

Before:
<img width="285" height="626" alt="Screenshot 2025-11-25 at 14 56 23" src="https://github.com/user-attachments/assets/fc7b19fe-1efa-4797-9267-85ede5b36688" />

After:
<img width="278" height="618" alt="Screenshot 2025-11-25 at 14 53 40" src="https://github.com/user-attachments/assets/e041799d-c26d-4466-b162-e2d4d2dd19e0" />

<!--

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., Talkback)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

<!--

Use these emoji to improve the code review

❤️ - Praise. Compliments about great solution.
⛏ - Nitpick️. Not high importance. Usually, short one-liners.
💡 - Idea. Alternative solution or improvement.
❓ - Question. Requires some explanation or more context.
⚠️ - Warning. Should be addressed in this PR or in some follow-up PR.
🛑 - Blocker. Must be addressed before PR gets merged.

-->

adalpari committed:
Replacing the old empty state illustration

adalpari committed:
Adding tags icon

dcalhoun commented on WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt:
```diff
@@ -512,13 +512,14 @@ class ReaderDiscoverViewModel @Inject constructor(
                 override val titleResId = R.string.reader_discover_empty_title
                 override val subTitleRes = R.string.reader_discover_empty_subtitle_follow
                 override val buttonResId = R.string.reader_discover_empty_button_text
+                override val illustrationResId = R.drawable.ic_reader_tag_96dp
```
From my testing, this appears unused. When navigating to _Your Tags_, all I see is the text and CTA button. Does this match your experience and expectation? 

<details><summary>Empty tags screenshot</summary>

![Image](https://github.com/user-attachments/assets/372100c1-6be6-42e2-bb35-65e4ab43a7a3)

</details>

adalpari replied:
This is actually a different scenario. I think I'm reverting it, because the icon is not providing any extra value.

dcalhoun commented on WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.kt:
```diff
@@ -2355,12 +2355,12 @@ class ReaderPostListFragment : ViewPagerFragment(), OnPostSelectedListener, OnFo
                             setEmptyTitleAndDescriptionForBookmarksList()
                             actionableEmptyView!!.image.setImageResource(
-                                R.drawable.illustration_reader_empty
+                                R.drawable.ic_reader_empty_state_96dp
```
The use of the search icon feels a bit odd to me for the empty bookmarks states. Something like [bookmark stacks](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:bookmark_stacks:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=bookmark&icon.size=24&icon.color=%231f1f1f) feels more appropriate. WDYT? 

<details><summary>Empty bookmarks screenshot</summary>

![Image](https://github.com/user-attachments/assets/0cc781e6-a88a-4545-9163-c2420db7ca0a)

</details>

It's unclear to me what Material Design icons we have access to/can use. E.g., why doesn't the search icon in the screenshot match the one I see in the [Material Symbols & Icon](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:search:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=search&icon.size=24&icon.color=%231f1f1f) search; our icon appears notched where the handle joins the circular glass. What is the official source we use for populating the files in `res/drawable`?

adalpari replied:
Good catch! I replaced all the empty states, but it doesn't fit here

dcalhoun commented on WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.kt:
```diff
@@ -2355,12 +2355,12 @@ class ReaderPostListFragment : ViewPagerFragment(), OnPostSelectedListener, OnFo
                         } else if ((currentTag?.isListTopic() == true) && isPostAdapterEmpty() && isAdded) {
                             actionableEmptyView!!.image.setImageResource(
-                                R.drawable.illustration_reader_empty
+                                R.drawable.ic_reader_empty_state_96dp
```
The use of the search icon feels a bit odd to me for the empty reader states. The accompanying text is also cryptic to me—what is the "filter button?" 

<details><summary>Empty subscriptions screenshot</summary>

![Image](https://github.com/user-attachments/assets/2b32742a-013b-4017-bbd4-ca06f46bf453)

</details>

I wonder if something like [article](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:article:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=article&icon.size=24&icon.color=%231f1f1f) is more appropriate. WDYT?

adalpari replied:
Let me have a look, but I think that probably using a generic icon is not adding any extra value. Probably changing the text will have a higher impact.

adalpari committed:
Removing images and adding a more descriptive texts and ctas

adalpari committed:
Deleting vectors

adalpari committed:
Merge branch 'trunk' into fix/CMM-988-Android-empty-states-use-outdated-visual-styles

adalpari committed:
Deleting the right file

adalpari committed:
Merge branch 'fix/CMM-988-Android-empty-states-use-outdated-visual-styles' of https://github.com/wordpress-mobile/WordPress-Android into fix/CMM-988-Android-empty-states-use-outdated-visual-styles

adalpari commented:
I've not changed the approach. I've modified the phrasing and added a CTA (or changed it) for better user understanding.
I don't think the plain icon is adding any UX benefit.

@dcalhoun wdyt?

<img width="361" height="792" alt="Screenshot 2025-11-25 at 16 29 17" src="https://github.com/user-attachments/assets/58cedc29-2e11-4bb4-9019-4e5c1cefe526" />
<img width="353" height="786" alt="Screenshot 2025-11-25 at 16 29 28" src="https://github.com/user-attachments/assets/97644aec-3875-49cc-9210-74e6728a4311" />

dcalhoun commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -2397,6 +2397,7 @@
     <string name="reader_empty_search_title">No results found</string>
     <string name="reader_empty_search_description">No results found for %s for your language</string>
     <string name="reader_empty_search_request_failed">Unable to perform search</string>
+    <string name="reader_empty_subscriptions">You haven\'t subscribed to any blog yet</string>ee
```
Update to remove erroneous trailing `ee` characters and improve clarity of the statement. 
```suggestion
    <string name="reader_empty_subscriptions">You haven\'t subscribed to any blogs yet</string>
```

adalpari replied:
Ups. Fixed!

dcalhoun reviewed (changes requested):
The strings file needs updating to remove erroneous syntax—see the inline comment. 

> I've modified the phrasing and added a CTA (or changed it) for better user understanding.
> I don't think the plain icon is adding any UX benefit.

Agreed. The latest approach is an improvement. The new CTA button also fixes CMM-995, the bug I noted earlier in https://github.com/wordpress-mobile/WordPress-Android/pull/22367#issuecomment-3576127744.

adalpari committed:
typo

adalpari committed:
typo
